util/readpass_file.c

```diff
@@ -0,0 +1,78 @@
+#include <stdio.h>
+#include <string.h>
+
+#include "insecure_memzero.h"
+#include "warnp.h"
+
+#include "readpass.h"
+
+/* Maximum file length. */
+#define MAXPASSLEN 2048
+
+/**
+ * readpass_file(passwd, filename):
+ * Read a passphrase from ${filename} and return it as a malloced
+ * NUL-terminated string via ${passwd}.  Produce an error if the file is 2048
+ * characters or more, or if it contains any newline \n or \r\n characters
+ * other than at the end of the file.  Do not include the \n or \r\n
+ * characters in the passphrase.
+ */
+int
+readpass_file(char ** passwd, const char * filename)
+{
+	FILE * fp;
```

Comment:
Why `fp`?  Normally we just call a file `f`...

Reply:
file pointer?  I hate one-character variables other than `i` and `j`.

... that said, I admit that you like `f` so I should follow that.

Reply:
Also, easier for searching -- not many words contain `fp`, but tons contain `f`.